### PR TITLE
feat: test and fixes

### PR DESCRIPTION
**Pull request by rebelchris**
In this PR we are fixing some edge-cases and re-evaluated the test cases we had.
The cases actually seem significant enough.

Decided to add a mobile render check for the sidebar to see if we are not breaking the mobile version.

- 0553e88add3d8e92b87bba3bdb65f4fac00491eb -> Added more space according to SC for the mobile version
- 8e89d4eae50933ddb1fcf149bf8eb07d37c95c2f -> Mobile backdrop was visible on desktop if you expand from mobile to desktop
- 275a38f96049cf01f1b939d8e789a9101919891b -> Reading history alignment was not set
- 22473237770ba8bee92b8aa86b3f881e0e5aaa2c -> Added a autoFocus attribute for the postSearch component
- 3c74a7d9fee0918135bc5b82245f13222a6e363f -> Added the mobile render test case for the sidebar

DD-350 #done

### CODE REVIEW

File: packages/shared/src/components/PostsSearch.tsx [hunk 1]
```diff
@@ -105,9 +107,11 @@ export default function PostsSearch({
...
   };
 
-  useEffect(() => {
-    searchBoxRef.current?.querySelector('input').focus();
-  }, [searchBoxRef]);
+  if (autoFocus) {
+    useEffect(() => {
+      searchBoxRef.current?.querySelector('input').focus();
+    }, [searchBoxRef]);
+  }
```
Comment (idoshamun) [inline packages/shared/src/components/PostsSearch.tsx:114]: Not sure conditional hooks are allowed in React.
It's better to put the condition inside the `useEffect`. I know it can mess their internals

File: packages/shared/src/components/sidebar/Sidebar.spec.tsx [hunk 2]
```diff
@@ -162,3 +167,11 @@ it('should set all navigation urls', async () => {
     );
   });
 });
+
+it('should render the mobile sidebar version on small screens', async () => {
+  await resizeWindow(1019, 768);
```
Comment (idoshamun) [inline packages/shared/src/components/sidebar/Sidebar.spec.tsx:172]: Nice trick!

**Review by idoshamun — CHANGES_REQUESTED** [review]
One change and we're ready to go!

**Comment by rebelchris** [comment]
@idoshamun Oh right!
Refactored to evaluate inside the `useEffect` hook

**Review by idoshamun — APPROVED** [review]
Woohoo!
